proto/skaffold.proto

```diff
@@ -231,4 +231,25 @@ enum ErrorCode {
 //    COULD_NOT_GET_DOCKER_CLIENT = 7;
 //    COULD_NOT_GET_LOCAL_CLUSTER = 8;
 //    BUILDER_CLEANUP = 9;
+
```

Comment:
I think we could follow grouping by first digit similar to HTTP status codes (https://www.rfc-editor.org/rfc/rfc7231.txt) 
I like the container errors being 3xx, and k8s infra errors 4xx, but then NO_ERROR should be 2xx, unknown 5xx, e.g. (unknown 500, unknown unsched 501) WDYT?

Reply:
sounds good. i have tried to group them, but not to the best!